lightning/src/chain/chainmonitor.rs

```diff
@@ -261,7 +261,14 @@ where C::Target: chain::Filter,
 	{
 		let mut dependent_txdata = Vec::new();
 		if let Some(height) = best_height {
-			self.highest_chain_height.fetch_max(height as usize, Ordering::AcqRel);
+			// Sadly AtomicUsize::fetch_max wasn't stabilized until 1.45, so until then we have to
+			// manually CAS.
+			let mut old_height = self.highest_chain_height.load(Ordering::Relaxed);
+			while self.highest_chain_height
+				.compare_exchange(old_height, height as usize, Ordering::AcqRel, Ordering::Relaxed).is_err()
+			{
+				old_height = self.highest_chain_height.load(Ordering::Acquire);
+			}
```

Comment:
So taking a closer look, we actually never call `process_chain_data` when blocks are disconnected, so `highest_chain_height` won't be reduced. I suppose a user could call `best_block_updated` with a smaller height, though, and there's some logic in `ChannelMonitor` to handle this case.

But when given a smaller height, won't this not be equivalent to `fetch_max`?